docs/abi.rst

```diff
@@ -0,0 +1,652 @@
+
+Because of the difference in complexity and efficiency when working with static and dynamic types, **we strongly recommend using static types over dynamic types whenever possible**.
+
+Instantiating Types
+^^^^^^^^^^^^^^^^^^^
+
+There are a few ways to create an instance of an ABI type. Each method produces the same result, but some may be more convenient than others.
+
+With the Constructor
+""""""""""""""""""""""
+
+The most obvious way is to use its constructor, like so:
+
+.. code-block:: python
+
+    myUint8 = abi.Uint8()
+    myUint64 = abi.Uint64()
+    myArrayOf12Uint8s = abi.StaticArray(abi.StaticArrayTypeSpec(abi.Uint8TypeSpec(), 12))
+
+For simple types, using the constructor is straightforward and works as you would expect. However, more complex types like :any:`abi.StaticArray` have type-level arguments, so their constructor must take an :any:`abi.TypeSpec` which fully defines all necessary arguments. These types can be created with a constructor, but it's often not the most convenient way to do so.
+
+With an :code:`abi.TypeSpec` Instance
+""""""""""""""""""""""""""""""""""""""
+
+You may remember that :code:`abi.TypeSpec` has a :any:`new_instance() <abi.TypeSpec.new_instance>` method that can be used to instantiate ABI types. This is another way of instantiating ABI types, if you happen to have an :code:`abi.TypeSpec` instance available. For example:
+
+.. code-block:: python
+
+    myUintType = abi.Uint8TypeSpec()
+    myUint8 = myUintType.new_instance()
+
+    myArrayType = abi.StaticArrayTypeSpec(myUintType, 12)
+    myArrayOf12Uint8s = myArrayType.new_instance()
+
+With :code:`abi.make`
+"""""""""""""""""""""
+
+Using :code:`abi.TypeSpec.new_instance()` makes sense if you already have an instance of the right :code:`abi.TypeSpec`, but otherwise it's not much better than using the constructor. Because of this, we have the :any:`abi.make` method, which is perhaps the most convenient way to create a complex type.
+
+To use it, you pass in a Python type annotation that describes the ABI type, and :code:`abi.make` will create an instance of it for you. For example:
+
+.. code-block:: python
+
+    from typing import Literal
+
+    myUint8 = abi.make(abi.Uint8)
+    myUint64 = abi.make(abi.Uint64)
+    myArrayOf12Uint8s = abi.make(abi.StaticArray[abi.Uint8, Literal[12]])
+
+.. note::
+    Since Python does not allow integers to be directly embedded in type annotations, you must wrap any integer arguments in the :code:`Literal` annotation from the :code:`typing` module.
+
+Computed Values
+^^^^^^^^^^^^^^^^^^^^^^^
+
+With the introduction of ABI types, it's only natural for there to be functions and operations which return ABI values. In a conventional language, it would be enough to return an instance of the ABI type directly from the operation, as usual. However, in PyTeal, there operations must actually return two things:
```

Comment:
@jasonpaulos Few _suggestions_ here:
* Remove _there_ - I think it's a typo.  Possibly, another phrasing was intended.
* Optionally, I'd opt for a more specific term (values) rather than _things_.

```suggestion
With the introduction of ABI types, it's only natural for there to be functions and operations which return ABI values. In a conventional language, it would be enough to return an instance of the ABI type directly from the operation, as usual. However, in PyTeal, operations must actually return two values:
```